.github/workflows/maven.yml

```
@@ -0,0 +1,23 @@
+name: Thunder CI pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
```

There is a newer version v4 available for setup-java action. I suggest using that instead for latest 
```suggestion
        uses: actions/setup-java@v4
```